service/nodeconfig/v1/resource/node/create.go

```diff
@@ -83,35 +82,26 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 
 		for _, p := range podList.Items {
 			if p.GetNamespace() == "kube-system" {
-				fmt.Printf("systemPods: %#v\n", p.GetName())
 				systemPods = append(systemPods, p)
 			} else {
-				fmt.Printf("customPods: %#v\n", p.GetName())
 				customPods = append(customPods, p)
 			}
 		}
 	}
 
-	{
-		fmt.Printf("\n")
-		fmt.Printf("customPods\n")
-		for _, p := range customPods {
-			fmt.Printf("%#v\n", p)
+	for _, p := range customPods {
+		err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
+		if err != nil {
+			return microerror.Mask(err)
 		}
-		fmt.Printf("customPods\n")
-		fmt.Printf("\n")
+	}
 
-		fmt.Printf("\n")
-		fmt.Printf("systemPods\n")
-		for _, p := range systemPods {
-			fmt.Printf("%#v\n", p)
+	for _, p := range systemPods {
+		err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
+		if err != nil {
+			return microerror.Mask(err)
 		}
-		fmt.Printf("systemPods\n")
-		fmt.Printf("\n")
 	}
```

Comment:
We delete the system pods next. I noticed that the calico and kube proxy pods are just launched again on the cordoned machine. I think this is why `kubectl` just excludes them when draining a node. Anyway I think this is good because it means all traffic being left behind is cut off regardless and we have a chance that even that might be graceful depending on what calico and the kube proxy is doing.

Reply:
I think this sounds about right (regarding the calico & kube-proxy).